endpoints/setuid.go

```diff
@@ -164,55 +169,135 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 }
 
 // extractGDPRInfo looks for the GDPR consent string and GDPR signal in the GPP query params
-// first and the 'gdpr' and 'gdpr_consent' query params second.
-func extractGDPRInfo(query url.Values) (gdpr.RequestInfo, error) {
-	// Defualt GDPR values
-	gdprConsent := query.Get("gdpr_consent")
-	gdprSignal := gdpr.SignalAmbiguous
+// first and the 'gdpr' and 'gdpr_consent' query params second. If found in both, throws a
+// warning. Can also throw a parsing or validation error
+func extractGDPRInfo(query url.Values) (reqInfo gdpr.RequestInfo, err error) {
+
+	reqInfo, err = parseGDPRFromGPP(query)
+	if err != nil {
+		return gdpr.RequestInfo{GDPRSignal: gdpr.SignalAmbiguous}, err
+	}
+
+	legacySignal, legacyConsent, err := parseLegacyGDPRFields(query, reqInfo.GDPRSignal, reqInfo.Consent)
+	isWarning := errortypes.IsWarning(err)
+	// If not warning, throw error
```

Comment:
I think this comment can be deleted since the flow of the code explains what it's doing nicely.

Reply:
Removed